src/main/java/com/minecolonies/coremod/colony/buildings/workerbuildings/BuildingFlorist.java

```diff
@@ -149,15 +150,14 @@ public void removePlantableGround(final BlockPos pos)
     public ItemStack getFlowerToGrow()
     {
         final List<ItemStorage> stacks = getPlantablesForBuildingLevel(getBuildingLevel()).stream()
-                                           .filter(stack -> !getModuleMatching(ItemListModule.class, m -> m.getId().equals(BUILDING_FLOWER_LIST)).isItemInList(stack))
-                                           .collect(Collectors.toList());
+          .filter(stack -> !getModuleMatching(ItemListModule.class, m -> m.getId().equals(BUILDING_FLOWER_LIST)).isItemInList(stack)).toList();
+
         if (stacks.isEmpty())
         {
             return null;
         }
 
-        Collections.shuffle(stacks);
-        return stacks.get(0).getItemStack();
+        return stacks.get(MathUtils.RANDOM.nextInt(stacks.size())).getItemStack();
```

Comment:
You can pull a random in one call, instead of copying it into a list...

Adding:
`.sorted((o1, o2) -> ThreadLocalRandom.current().nextInt(-1, 2))
            .findAny()
`
in the stream chain will return an entry without putting it in the list first.

Reply:
That is code that is not obvious to the reader and has even a chance to perform worse than this. I did research quite a lot about random elements from a stream and it's not super reliable.

Reply:
True, i wish there was a better approach to this